### PR TITLE
Only upgrade necessary dependencies in brew upgrade. Fixes #1178

### LINKED ISSUE #1178: [osx] Don't blanket upgrade brew dependancies in configure.sh
Instead of running brew upgrade to upgrade every installed dependancy, only upgrade those required by OpenBazaar. i.e python gpg sqlite3 wget openssl zmq autoenv

<bountysource-plugin>

---
Want to back this issue? **[Place a bounty on it!](https://www.bountysource.com/issues/8208758-osx-don-t-blanket-upgrade-brew-dependancies-in-configure-sh?utm_campaign=plugin&utm_content=tracker%2F1516753&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1516753&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by fanquake**
Rather than running ‘brew upgrade’ which will upgrade every package
installed on the system, only upgrade those that we need.

### CODE REVIEW

**Comment by hoffmabc** [comment]
Leaving note to remove zmq from this.

File: configure.sh [hunk 1]
```diff
@@ -35,12 +35,13 @@ function brewDoctor {
 
 function brewUpgrade {
     echo "If your homebrew packages are outdated, we recommend upgrading them now. This may take some time."
+    echo "We will only upgrade the packages needed to run OpenBazaar"
     read -r -p "Do you want to do this? [y/N] " response
     if [[ $response =~ ^([yY][eE][sS]|[yY])$ ]]
     then
-      if ! brew upgrade; then
+      if ! brew upgrade autoenv gpg openssl sqlite3 wget zmq; then
```
Comment (dionyziz) [inline configure.sh]: DRY

**Comment by fanquake** [comment]
Removed zmq

**Comment by hoffmabc** [comment]
Thanks

**Comment by hoffmabc** [comment]
Can you please rebase and check it out again?

**Comment by fanquake** [comment]
Rebased

**Comment by hoffmabc** [comment]
Thanks!

**Pull request merged**